File: CPP solutions/binary_tree_postorder_traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode *root) {
		vector<int> result;
        if (!root) return result;
        traverse(result, root);
        return result;
    }
    void traverse(vector<int> &result, TreeNode *root)
    {
        if (root->left) traverse(result, root->left);
        if (root->right) traverse(result, root->right);
        result.push_back(root->val);
    }
};

File: Python solutions/binary_tree_postorder_traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    # @param root, a tree node
    # @return a list of integers

    def postorderTraversal(self, root):
        result = []
        if root:
            self.traverse(result, root)
        return result

    def traverse(self, result, root):
        if root.left is not None:
            self.traverse(result, root.left)
        if root.right is not None:
            self.traverse(result, root.right)
        result.append(root.val)
